### DISCUSSION TITLE
Progress Tracking

### DISCUSSION THREAD

**Discussion by sak345** [freeCodeCamp/Developer_Quiz_Site#858] [Ideas]
## Description:

Implement a progress tracking system that help users set goals and track their improvement as they go through the quiz questions. This feature can be particularly helpful for learners who want to systematically build their knowledge and skills in specific programming topics.

## Implementation:

- **User Profiles:** Create user profiles where users can set their learning goals, track their progress, and view their achievements.

- **Progress Tracking:** Users can mark questions as completed, and the platform should track their performance and the time spent on each question.

- **Achievement Badges:** Reward users with badges or achievements as they progress through their chosen learning paths. Achievements could be based on reaching specific milestones, mastering certain topics, or consistently answering questions correctly.

- **Recommendations:** Provide recommendations for the next questions or topics to study based on a user's performance and goals.

- **Community Leaderboards:** Implement leaderboards to encourage healthy competition among users and motivate them to excel in their chosen learning paths.

**Comment by jdwilkin4** [comment]
Hi @sak345 !

Thank you for your suggestions.

The team in the past has talked about similar changes like you are proposing but we have come to decision that we are not interested in significantly building out the app in this way at this time.
Right now the team's priority is with expanding the current curriculum on the main freeCodeCamp learn page.

Once those larger projects are out of the way, then could circle back and open this up for discussion.
But for now, we are not going to be pursuing this path at this time.

Thanks for your understanding.
